buffer/src/main/java/io/netty/buffer/PoolArena.java

```diff
@@ -727,8 +727,8 @@ protected void memoryCopy(ByteBuffer src, int srcOffset, PooledByteBuf<ByteBuffe
 
             if (HAS_UNSAFE) {
                 PlatformDependent.copyMemory(
-                        PlatformDependent.directBufferAddress(src) + srcOffset,
-                        PlatformDependent.directBufferAddress(dstBuf.memory) + dstBuf.offset, length);
+                        PlatformDependent.directBufferAddress(src).add(srcOffset),
+                        PlatformDependent.directBufferAddress(dstBuf.memory).add(dstBuf.offset), length);
```

Comment:
call copy memory with the offset versuon, also cam you just call src.memoryAddress (please check this)

Reply:
The `src.memoryAddress()` method is package private so can't be called from here